File: src/types/note.ts
export interface WindowBounds{x:number;y:number;width:number;height:number}
export interface Note{schemaVersion:number;id:string;title:string;content:string;tags:string[];color:string;pinned:boolean;floating:boolean;collapsed:boolean;alwaysOnTop:boolean;windowBounds:WindowBounds|null;deleted:boolean;createdAt:string;updatedAt:string}
export type NotePatch=Partial<Omit<Note,"id"|"schemaVersion"|"createdAt"|"updatedAt">>

File: src-tauri/src/models/note.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct WindowBounds {
    pub x: f64,
    pub y: f64,
    pub width: f64,
    pub height: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct Note {
    pub schema_version: u32,
    pub id: String,
    pub title: String,
    pub content: String,
    pub tags: Vec<String>,
    pub color: String,
    pub pinned: bool,
    pub floating: bool,
    pub collapsed: bool,
    pub always_on_top: bool,
    pub window_bounds: Option<WindowBounds>,
    pub deleted: bool,
    pub created_at: String,
    pub updated_at: String,
}
